### PR TITLE
Add public_network_access_enabled to azurerm_redis_cache

### PR DESCRIPTION
**Pull request by mybayern1974**
To address issue #9307. Marking this as [WIP] for I'm still running regression acctest under Redis. Once things could pass I would remove the [WIP]. Below paste the run pass result for the newly added test case.
```
=== RUN   TestAccRedisCache_PublicNetworkAccessDisabledEnabled
=== PAUSE TestAccRedisCache_PublicNetworkAccessDisabledEnabled
=== CONT  TestAccRedisCache_PublicNetworkAccessDisabledEnabled
--- PASS: TestAccRedisCache_PublicNetworkAccessDisabledEnabled (1965.03s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: website/docs/r/redis_cache.html.markdown [hunk 1]
```diff
@@ -65,6 +65,8 @@ The following arguments are supported:
 
 * `private_static_ip_address` - (Optional) The Static IP Address to assign to the Redis Cache when hosted inside the Virtual Network. Changing this forces a new resource to be created.
 
+* `public_network_access_enabled` - (Optional) Whether or not public network access is allowed for this Redis Cache. Defaults to `true`.
```
Comment (katbyte) [inline website/docs/r/redis_cache.html.markdown:68]: Might be best to expand on what public network access means
```suggestion
* `public_network_access_enabled` - (Optional) Whether or not public network access to the internet is allowed for this Redis Cache. Defaults to `true`.
```

**Review by katbyte — APPROVED** [review]
Aside from one comment this LGTM 👍

File: website/docs/r/redis_cache.html.markdown (hunk 1 above)
Comment (mybayern1974) [reply website/docs/r/redis_cache.html.markdown]: Updated

**Comment by mybayern1974** [comment]
Thank you katbyte for your review. I've updated accordingly. 

As for regression acctest, all (totally 28) whose names start with "TestAccRedis..." have passed. So remove the [WIP] from the PR title.

**Pull request merged**